hivemq-edge/src/main/java/com/hivemq/api/resources/impl/ProtocolAdaptersResourceImpl.java

```diff
@@ -81,20 +71,28 @@ public ProtocolAdaptersResourceImpl(
     public @NotNull Response getAdapterTypes() {
 
         //-- Obtain the adapters installed by the runtime (these will be marked as installed = true).
-        Set<ProtocolAdapter> installedAdapters = protocolAdapterManager.getAllAvailableAdapterTypes().
-                values().
-                stream().
-                map(installedAdapter -> ProtocolAdapterApiUtils.convertInstalledAdapterType(
-                        objectMapper, protocolAdapterManager, installedAdapter, configurationService)).
-                collect(Collectors.toSet());
+        Set<ProtocolAdapter> installedAdapters =
+                protocolAdapterManager.getAllAvailableAdapterTypes().values().stream().map(installedAdapter -> {
+                    try {
+                        return ProtocolAdapterApiUtils.convertInstalledAdapterType(objectMapper,
+                                protocolAdapterManager,
+                                installedAdapter,
+                                configurationService);
+                    } catch (Throwable t) {
+                        logger.warn("Not able to properly load protocol adapter", t);
+                        return null;
```

Comment:
What happens if you return a null in a map() function?

Reply:
the stream then contains a null value at this place, that get's then removed by the `filter(Objects::nonNull)` afterwards